src/test/java/org/sagebionetworks/bridge/spring/util/EtagComponentTest.java

```
@@ -264,4 +298,112 @@ public void nullBridgeSessionHeader() throws Throwable {
         
         component.checkEtag(mockJoinPoint);
     }
+    
+    @Test
+    public void invalidateCacheOnChange_invalidAttributeNoop() throws Throwable {
+        EtagCacheKey badTimeZoneAnn = new EtagCacheKey() {
+            @Override
+            public Class<? extends Annotation> annotationType() {
+                return EtagCacheKey.class;
+            }
+            @Override
+            public Class<?> model() {
+                return DateTimeZone.class;
+            }
+            @Override
+            public String[] keys() {
+                return new String[] {"userId"};
+            }
+            @Override
+            public String invalidateCacheOnChange() {
+                return "bad-value";
+            }
+        };
+        // This just gets ignored
+        doReturn(ImmutableList.of(badTimeZoneAnn)).when(mockContext).getCacheKeys();
+
+        when(mockRequest.getHeader(IF_NONE_MATCH)).thenReturn(ETAG);
+        
+        component.checkEtag(mockJoinPoint);
+        
+        verify(mockCacheProvider, never()).removeObject(any());
+    }
```

Maybe this should throw an exception-this would be a code configuration error.

---

This now throws an IllegalArgumentException